src/core/shape_inference/include/sparse_fill_empty_rows_shape_inference.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <unordered_set>
+
+#include "openvino/op/sparse_fill_empty_rows.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v16 {
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const SparseFillEmptyRows* op,
+                                 const std::vector<TShape>& input_shapes,
+                                 const ITensorAccessor& tensor_accessor = make_tensor_accessor()) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 4);
+
+    const auto& values_shape = input_shapes[0];
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           values_shape.rank().compatible(1),
+                           "The values input must be a 1D tensor. Got: ",
+                           values_shape);
```

Comment:
```suggestion
    NODE_SHAPE_INFER_CHECK(op,
                           input_shapes,
                           input_shapes[0].rank().compatible(1),
                           "The values input must be a 1D tensor.");
```
Or try remove the `values_shape` as all input shapes will be in error message. Or explain in error message at which shape look to check.

Same for other input shapes checks.

Reply:
Done, I simplified the error messages.